tests/nvmeof/test_io_perf.py

```diff
@@ -456,32 +456,46 @@ def nvmeof(ceph_cluster, **args):
     rbd = args["rbd"]
 
     # Configure NVMEoF Gateway
-    gw_node = get_node_by_id(ceph_cluster, args["gw_node"])
-    NVMeGWCLI.NVMEOF_CLI_IMAGE = cli_image
-    nvmegwcli = NVMeGWCLI(gw_node, 5500)
     initiator = get_node_by_id(ceph_cluster, args["initiator_node"])
     initiator.exec_command(cmd=f"mkdir -p {ARTIFACTS_DIR}", sudo=True)
 
     try:
         # Install Gateway
-        deploy_nvme_service(ceph_cluster, args)
+        nvme_service = NVMeService(args, ceph_cluster)
 
         for count in range(args["iterations"]):
-            name = generate_unique_id(4)
             subsystem = {
-                "nqn": f"nqn.2016-06.io.spdk:{generate_unique_id(4)}",
+                "nqn": args["subsystems"][0]["nqn"],
                 "serial": generate_unique_id(2),
                 "allow_host": "*",
-                "listener_port": find_free_port(gw_node),
+                "listener_port": args["subsystems"][0]["listener_port"],
                 "node": args["gw_node"],
             }
+
             initiator_cfg = {
                 "subnqn": subsystem["nqn"],
                 "listener_port": subsystem["listener_port"],
                 "node": args["initiator_node"],
             }
-            configure_subsystems(ceph_cluster, rbd, pool, nvmegwcli, subsystem)
+            # Deploy NVMe Service
+            if args.get("install"):
+                nvme_service.deploy()
+
+            # Initialize gateways
+            nvme_service.init_gateways()
+            nvmegwcli = nvme_service.gateways[0]
+
+            # Configure Subsystems
+            configure_subsystems(nvme_service)
```

Comment:
Can we not directly call configure_gw_entities and that will take care of configuring subsystems, listeners and hosts right? All you need to do is pass those keywords to config I guess. Can we check this?